### DISCUSSION TITLE
Command Flags

### DISCUSSION THREAD

**Discussion by Moburma** [swfans/syndwarsfx#227] [General]
I realised some of the flags are now documented in command.h, I thought I would share ones I know of, and also ones I know specific examples of, but don't necessarily understand the overall intended wider purpose of. I started writing some of these up a while ago for my level editor, but makes sense to put some of them in the actual code.

PCmdF_Unkn0008 = 0x0008  - Not 100% on the wider use of this, but it definitely is used to stop CAMERA_TRACK commands. To stop the camera tracking an NPC, you need to call the command again, but this time with this flag set.

 PCmdF_Unkn0010 = 0x0010 - Rectangular area mode. This works with a few different commands (most notably WITHIN_AREA or the various *_ARRIVE commands) to change them from circular radius mode (where Arg1 determines the radius of the area needed to enter) to rectangle mode, where the top left corner of the rectangle is determined by coordinates in X and Z, and the bottom right corner by those in Arg1 and Time.

**Comment by mefistotelis** [comment]
Both these flags are now named and commented:
https://github.com/swfans/swars/blob/81d3f16fccfde3911af5eb909abe78dacd880149/src/command.h#L227

**Comment by Moburma** [comment]
I also noticed there is an interesting use of PCmdF_RevertFunct on the level C007L007.D1 that works as a NOT. The level starts off with several agents effectively surrounding the player, but they begin far off. The agents have a KILL_ALL_GROUP command to kill the player group, but only after a WAIT_MEM_G_ARRIVE command. This wait command is set with flag 8. As the player starts _inside_ this trigger and it is set to flag 8, it has interesting behaviour. It's set to check for 1 member of group 0, but actually what it does is only trigger when there is _not_ even 1 member of the group in the trigger area. As such it doesn't trigger the kill command for the agents until they all unguided have left the WAIT command trigger area.

**Reply by mefistotelis** [reply]
Right, the use with "WAIT" commands should be mentioned in documentation of the flag.
